src/test/java/test/listeners/ExecutionListenerAndSuiteListenerTest.java

```diff
@@ -0,0 +1,39 @@
+package test.listeners;
+
+import static org.testng.Assert.assertEquals;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.testng.TestNG;
+import org.testng.TestNGException;
+import org.testng.annotations.Test;
+import org.testng.xml.Parser;
+import org.testng.xml.XmlSuite;
+import org.xml.sax.SAXException;
+
+
+public class ExecutionListenerAndSuiteListenerTest {
+
+    @Test
+    public void executionListenerAndSuiteListenerTest() {
+        String suiteFile = "src/test/resources/executionlistenersingletoncheck/parent.xml";
+        runTests(suiteFile);
+    }
+
+    private void runTests(String suiteFile) {
+        List<XmlSuite> suites;
+        try {
+            suites = new Parser(suiteFile).parseToList();
+        } catch (ParserConfigurationException | SAXException | IOException e) {
+            throw new TestNGException(e);
+        }
+        TestNG testng = new TestNG();
+        testng.setXmlSuites(suites);
+        ExecutionListenerAndSuiteListener listener = new ExecutionListenerAndSuiteListener();
```

Comment:
It is unexpected because most of the time we add a listener instance into testng which is not the case here.
And when we look deeper, we find you just looking for a static var.

Reply:
And when we look deeper, we find you just looking for a static var.

Sorry, I cannot follow you. What do you mean?

Reply:
You create an instance to get the value of a static var. I don't understand the logic.

Reply:
aha, you are right. My logic here is bad. will fix it.